### DISCUSSION TITLE
屏蔽特定动态

### DISCUSSION THREAD

**Discussion by YaeSakuraQvQ** [the1812/Bilibili-Evolved#4541] [提问 / Q&A] [answered]
比如我想屏蔽转发的互动抽奖动态，我该如何做
看着网上的教程写正则，填到动态过滤里都没生效

**Comment by the1812** [comment]
你希望的规则是什么, 匹配某一段文本?

**Comment by YaeSakuraQvQ** [comment]
<img width="317" alt="image" src="https://github.com/the1812/Bilibili-Evolved/assets/87939280/6a1d2e86-b545-483d-8730-782f60e50080">

比如像这样
转发的时候文字是默认的"转发动态" 然后下面含有"互动抽奖"

**Reply by the1812** [reply]
试试这个: `/转发动态\s*互动抽奖/`

**Reply by YaeSakuraQvQ** [reply]
无效
单独填入"转发动态"或"互动抽奖"皆可进行屏蔽（但那样误伤太广了，希望可以精准匹配）

**Reply by YaeSakuraQvQ** [reply]
我又试了一下，上次我应该是刚好遇到无法匹配的情况了，这个正则可以匹配大多数，但是还是有些不足。
如`互动抽奖`后面没加空格而是直接跟内容的无法匹配：
`互动抽奖 测试测试测试` 可以被匹配
`互动抽奖测试测试测试` 无法被匹配
还有一些以其他形式出现的无法被匹配，如`#互动抽奖#`
非常感谢大佬了，如果有时间可以再完善一下就更好了~~

**Reply by the1812** [reply, marked as answer]
其他各种格式只能你自己写正则了, 是否匹配可以在 https://regex101.com/ 里看, 当前页面动态卡片的数据可以在浏览器控制台里看
```
console.log(bilibiliEvolved.componentApis.feeds.api.feedsCardsManager.cards)
```
![Snipaste_2024-01-14_15-02-46](https://github.com/the1812/Bilibili-Evolved/assets/26504152/216054b1-2669-4009-991b-5ecd31c542a6)

里面的 `text` 就是动态的文字, `repostText` 是其中被转发部分的文字

**Reply by YaeSakuraQvQ** [reply]
感谢大佬，我再自己试试叭

**Reply by YaeSakuraQvQ** [reply]
我又试了多种形式的正则，还是效果不佳。
而且有的情况控制台输出与实际文本显示不符，也没匹配到，如：
![image](https://github.com/the1812/Bilibili-Evolved/assets/87939280/dd4c7d27-7985-4427-9116-62a284e4bf74)
![image](https://github.com/the1812/Bilibili-Evolved/assets/87939280/1bd788a5-ea77-497b-b868-4be7b6eec627)
有没有可能先去直接识别这种特殊动态，然后再通过正则匹配"转发动态"这四个字
比如评论中或者动态中up带货那种格式和抽奖这种格式，b站都会做不同的显示，我推测其可能为一种特殊格式
![image](https://github.com/the1812/Bilibili-Evolved/assets/87939280/460dfcf6-acf3-48f7-8a68-62ba906998f5)
